internal/datastore/postgres/postgres_test.go

```diff
@@ -550,209 +543,6 @@ func QuantizedRevisionTest(t *testing.T, b testdatastore.RunningEngineForTest) {
 	}
 }
 
-func XIDMigrationAssumptionsTest(t *testing.T, b testdatastore.RunningEngineForTest) {
```

Comment:
@jakedt would love to hear your thoughts on this. I've removed this test. I'll just quote what I wrote in the commit log:

>unfortunately this test cannot be held around as the postgresql datastore evolves. It's the only test that runs code at a specific set of revisions, and therefore any changes to the code would have to maintain backward compatibility w.r.t to those revisions.

>In this particular example `QueryRelationships` failed to find the new caveat columns. It would require us to maintain both execution paths: relationship query with and without caveats.

Reply:
We can't remove this test until a revision of SpiceDB is cut that includes the 4-phase migrations and we have migrated all of our managed environments successfully. The specific set of revisions are actually created manually, so there is nothing dangerous about that particular part of the migration. We haven't cut a release of this yet, so there is still time to slot your migration in ahead of the 4-phase migrations if you want. That should eliminate the issues you saw.

Reply:
@jakedt I've inserted the caveat migration right before the `xid` migrations and the removed test is back and 🟢